eox_core/api/v1/api_schema.py

```diff
@@ -0,0 +1,59 @@
+"""
+Swagger view generator
+"""
+from django.conf import settings
+from django.conf.urls import url
+from drf_yasg.generators import OpenAPISchemaGenerator
+from drf_yasg.openapi import SwaggerDict
+from drf_yasg.views import get_schema_view
+from edx_api_doc_tools import get_docs_cache_timeout, make_api_info
+from rest_framework import permissions
+
+from eox_core.api.v1 import views
+
+
+class APISchemaGenerator(OpenAPISchemaGenerator):
+    """
+    Schema generator for eox-core.
+
+    Define an exclusive base path to perform requests for each endpoint and a
+    specific security definition using oauth without overwritting project wide
+    settings.
+    """
+
+    def get_schema(self, request=None, public=False):
+        schema = super().get_schema(request, public)
+        schema.base_path = '/eox-core/api/v1/'
+        return schema
+
+    def get_security_definitions(self):
+        security_definitions = {
+            'OAuth2': {
+                'flow': 'application',
+                'tokenUrl': '{}/oauth2/access_token/'.format(settings.LMS_ROOT_URL),
+                'type': 'oauth2',
+            },
+        }
+        security_definitions = SwaggerDict.as_odict(security_definitions)
+        return security_definitions
+
+
+api_urls = [  # pylint: disable=invalid-name
```

Comment:
More URLs could be added here later, I think we should move this to another file
What do you think?

Reply:
I'm no quite sure how to handle this one. For now the idea was to add paths as the documentation for each resource is completed. Eventually we'll end up having a subset of `urls.py`, I'm not too much of a fan having those duplicates.